nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/AbstractAWSProcessor.java

```diff
@@ -286,7 +286,7 @@ protected void initializeRegionAndEndpoint(ProcessContext context) {
             final String urlstr = StringUtils.trimToEmpty(context.getProperty(ENDPOINT_OVERRIDE).evaluateAttributeExpressions().getValue());
             if (!urlstr.isEmpty()) {
                 getLogger().info("Overriding endpoint with {}", new Object[]{urlstr});
-                this.client.setEndpoint(urlstr);
+                this.client.setEndpoint(urlstr, this.client.getServiceName(), this.region.getName());
```

Comment:
these values will always be present?  We dont have to guard from them being null/undefined/different than the API being called expects?

Reply:
Yep. They will be present always. We have made `Region` a `required` parameter so it will be there. The service name is provided by the AWSClient implementation in the AWS SDK. Gone through their implementation and verified that, it takes care of undefined and null service names i.e, it computes the service name again and returns, if it turns out to be null.

Reply:
@joewitt The patch has been tested by @Mermadi and it is working fine. Can we close this one?

Reply:
Using this.region.getName() doesn't make sense to me because the aws docs for 
public void setEndpoint(String endpoint, String serviceName, String regionId)
say
     * @param regionId
     *            The ID of the region in which this service resides AND the
     *            overriding region for signing purposes.
so the regionId is the same as the overriding region but using this.region is using the enum.

Reply:
Thanks @amcdonaldccri for pointing this out. I'll work on fixing it.